File: backend/app/schemas/article.py
from typing import Optional, Literal
from uuid import UUID
from pydantic import BaseModel


class ArticleCardResponse(BaseModel):
    id: UUID
    wikipedia_id: int
    title: str
    extract: str
    category: Optional[str] = None
    reading_time_minutes: int
    is_featured: bool
    image_url: Optional[str] = None
    language: str
    is_liked: bool = False
    is_bookmarked: bool = False
    fun_fact: Optional[str] = None

    class Config:
        from_attributes = True


class ArticleFullResponse(ArticleCardResponse):
    full_content: Optional[str] = None


class ArticleFeedbackRequest(BaseModel):
    feedback: Literal["like", "not_interested"]


class ShareRequest(BaseModel):
    platform: str

File: frontend/src/types/article.ts
export interface ArticleCard {
  id: string
  wikipediaId: number
  title: string
  extract: string
  category: string | null
  readingTimeMinutes: number
  isFeatured: boolean
  imageUrl?: string | null
  language: string
  isLiked: boolean
  isBookmarked: boolean
  funFact?: string | null
  url?: string | null
  // UI-only helpers (computed or from optimistic updates)
  likeCount?: number
}

export interface ArticleFull extends ArticleCard {
  fullContent?: string | null
  content?: string | null  // alias — some API versions return this key directly
}

// Cursor-paginated response from GET /api/v1/articles/feed
export interface PaginatedFeedResponse {
  items: ArticleCard[]
  nextCursor: string | null   // snake_case "next_cursor" transformed to camelCase by api.ts
  hasMore: boolean            // snake_case "has_more" transformed by api.ts
}

// Single item returned by GET /api/v1/search
// Note: id here is wikipedia_id (integer), not an internal UUID
export interface SearchResultItem {
  id: number
  title: string
  extract: string
  language: string
}

// Full response envelope from GET /api/v1/search
export interface SearchResponse {
  query: string
  items: SearchResultItem[]
  total: number
  language: string
}

/** @deprecated Use SearchResultItem / SearchResponse — kept to avoid breaking any stale imports */
export interface SearchArticle {
  id: string
  title: string
  extract: string
  category: string | null
  readingTimeMinutes: number
  language: string
}
